engine/src/main/java/org/terasology/reflection/copy/CopyStrategy.java

```diff
@@ -25,8 +25,9 @@
 
     /**
      * @param value The value to copy
+     * @param copyEntities Whether to make deep copies of EntityRefs that are owned by this object
      * @return A safe to use copy of the given value.
      */
-    T copy(T value);
+    T copy(T value, boolean copyEntities);
```

Comment:
I did it this way largely because most of the implementations (all but one actually) are either exactly the same for both values of copyEntities or just pass on the flag to a recursive call, therefore this method seemed a bit simpler.